dashboard/src/components/AppView/AppView.tsx

```diff
@@ -240,6 +273,50 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
   private deleteApp = (purge: boolean) => {
     return this.props.deleteApp(this.props.releaseName, this.props.namespace, purge);
   };
+
+  private filterByKind = (kind: string) => {
+    const kubeItems = _.pickBy(this.props.resources, r => {
```

Comment:
IMO this component shouldn't care about all the resources, the container should filter down the Secrets that are necessary for rendering here.

Reply:
mm, why? I mean, the container doesn't need to know what the view wants to use, isn't' it? Also the method `getResource` is generic, I would see it weird if the `resources` are pre-filtered.

Also I am thinking about using this approach to potentially display information of all the resources that an application is composed of (I think that may be convenient for issues like #860).

Reply:
AFAIU the whole idea of the container is to pass down relevant data to the component for it to render. I don't like the idea of doing filtering inside a component when it's not based on user input, if we know what a component needs then I think it should be filtered at either the reducer or the container level.

If what the component needs to show might change based on some user action inside the component, it might make sense to do filtering here, but even in that case it should go through the Redux flow and enter as an updated prop in the component.